src/main/java/bio/terra/common/flagsmith/FlagsmithService.java

```diff
@@ -0,0 +1,50 @@
+package bio.terra.common.flagsmith;
+
+import com.flagsmith.FlagsmithClient;
+import com.flagsmith.config.FlagsmithCacheConfig;
+import com.flagsmith.exceptions.FlagsmithClientError;
+import com.flagsmith.models.Flags;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class FlagsmithService {
+  private static final Logger LOGGER = LoggerFactory.getLogger(FlagsmithService.class);
+
+  @Autowired private final FlagsmithProperties flagsmithProperties;
+
+  FlagsmithService(FlagsmithProperties flagsmithProperties) {
+    this.flagsmithProperties = flagsmithProperties;
+  }
+
+  public boolean isFeatureEnabled(String feature, Optional<Boolean> defaultValue) {
```

Comment:
Minor: consider this signature

public Optional<Boolean> isFeatureEnabled(String feature) {
  // if flagsmith is not enabled, optional.emtpy()
}

reason : this library does not need to know the default value. Default behavior is determined by the client.

Reply:
good idea.